packages/common/core-utils/src/heap.ts

```diff
@@ -173,10 +177,14 @@ export class Heap<T> {
 	}
 
 	private swap(k: number, j: number): void {
-		const tmp = this.L[k];
-		this.L[k] = this.L[j];
-		this.L[k].position = k;
-		this.L[j] = tmp;
-		this.L[j].position = j;
+		// Non null assertig here since this affects perf if we assert
```

Comment:
"asserting" misspelled here and below. Anyway, I would suggest:
Array is always filled and indices are within range.

Reply:
Probably easiest to disable no-non-null-assertion for Heap implementation scope. That will address question about comments for above locations.